src/components/pages/Header.jsx

```diff
@@ -0,0 +1,57 @@
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+import DesktopNav from "../navbars/DesktopNav";
+import MobNav from "../navbars/MobNav";
+import ProfileNav from "../navbars/ProfileNav";
+
+const Header = () => {
+    const [isSearch, setIsSearch] = useState(true);
+    const [isLoginBtn, setIsLoginBtn] = useState(true);
+    const [isJoinBtn, setIsJoinBtn] = useState(true);
+    
+    const location = useLocation();
+    useEffect(() => {
+        if (location.pathname === "/search") {
```

Comment:
This is good but a little verbose. The logic here is pretty straightforward. Let's try a switch statement or ternary operators.